### PR TITLE
Upload/Download mot files functions

### PR DESCRIPTION
**Pull request by jgsdavies**

### CODE REVIEW

File: tests/test_rpc_client_core.py [hunk 1]
```diff
@@ -126,6 +127,27 @@ def test_rpc_communication_error():
         assert "One or more parameter types were invalid" in str(e_info.value)
 
 
+def test_ensure_version_later_than():
```
Comment (sarah-r-woodrow) [inline tests/test_rpc_client_core.py:130]: Is this test finished? Should have some asserts to give any results?

**Review by sarah-r-woodrow — CHANGES_REQUESTED** [review]

**Review by sarah-r-woodrow — CHANGES_REQUESTED** [review]

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 2]
```diff
@@ -458,3 +458,45 @@ def quit(self):
...
+
+        Parameters
+        ----------
+        file_path : str
+            Full path to the mot file, including the file name and .mot extension.
+            Use the ``r'filepath'`` syntax to force Python to ignore special characters.
+        """
+        self.connection.ensure_version_at_least("2023.2")
+        file_contents = []
+        with open(file_path, "r") as mot_file:
```
Comment (sarah-r-woodrow) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:496]: What if the mot file doesn't exist? Do we need to show an error to the user?

**Review by sarah-r-woodrow — CHANGES_REQUESTED** [review]

File: tests/test_rpc_client_core.py [hunk 3]
```diff
@@ -126,6 +127,27 @@ def test_rpc_communication_error():
...
+    mock_motorcad_connection.program_version = "2023.2.0"
+    mock_motorcad_connection.ensure_version_at_least("2022.1.2")
+
+    mock_motorcad_connection.program_version = "2023.1.2"
+    mock_motorcad_connection.ensure_version_at_least("2023.1.2")
+
+    mock_motorcad_connection.program_version = "2023.1.2.0"
+    mock_motorcad_connection.ensure_version_at_least("2023.1.2")
+
+    # with pytest.raises(MotorCADError):
```
[the remark this answers is not in the document]
Comment (jgsdavies) [reply tests/test_rpc_client_core.py:146]: I want to try and get it working at some point. I think it's a valid test but it's failing on the test server. Had a go doing this but didn't want to spend more time on it right now

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 4]
```diff
@@ -458,3 +458,45 @@ def quit(self):
...
+        file_contents = self.connection.send_and_receive(method)
+
+        with open(file_path, "w") as mot_file:
+            for line in file_contents:
+                mot_file.write(line + "\n")
+
+    def upload_mot_file(self, file_path):
+        """Upload a .mot file to Motor-CAD instance from local directory.
+
+        This allows users to send .mot files from a local directory to a Motor-CAD instance
```
[the remark this answers is not in the document]
Comment (jgsdavies) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py:485]: I've added the links to save/load file

File: src/ansys/motorcad/core/methods/rpc_methods_general.py (hunk 2 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py:496]: image

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 5]
```diff
@@ -458,3 +458,45 @@ def quit(self):
     def set_free(self):
         """Free the Motor-CAD instance."""
         return self.connection._set_free()
+
+    def download_mot_file(self, file_path):
```
Comment (ravi-ansys) [inline src/ansys/motorcad/core/methods/rpc_methods_general.py:462]: If the file i want to download has some error messages, which i okayed. However as soon as i use this funtion the same error message popup and process stops. image.png

File: src/ansys/motorcad/core/methods/rpc_methods_general.py [hunk 6]
```diff
@@ -458,3 +458,45 @@ def quit(self):
...
+        ----------
+        file_path : str
+            Full path to the mot file, including the file name and .mot extension.
+            Use the ``r'filepath'`` syntax to force Python to ignore special characters.
+        """
+        self.connection.ensure_version_at_least("2023.2")
+        method = "GetCurrentFileJSON"
+        file_contents = self.connection.send_and_receive(method)
+
+        with open(file_path, "w") as mot_file:
```
[the remark this answers is not in the document]
Comment (sarah-r-woodrow) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py:478]: Ah, good!

File: tests/test_rpc_client_core.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sarah-r-woodrow) [reply tests/test_rpc_client_core.py:148]: Okay, sounds good. Thanks for the comment.

**Review by sarah-r-woodrow — APPROVED** [review]

File: src/ansys/motorcad/core/methods/rpc_methods_general.py (hunk 5 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/methods/rpc_methods_general.py:462]: Hi @ravi-ansys, I think this is just a Motor-CAD warning. This works fine when MessageDisplayState=2

**Pull request merged**